### ISSUE TITLE
Implement Qt styling for Avalon tools

### ISSUE THREAD

**Issue by BigRoy** [getavalon/core#293]
# Issue

Currently we've been enjoying the Avalon tools most of the time in Maya only where they took over the Maya styling and looked fine. Outside of Maya though they use the default OS styling and seem to be lacking the good looks that we want, and in some cases it even becomes a somewhat unreadable mess (like in `cbloader` currently)

Let's set up a Qt stylesheet for Avalon to always look great no matter the circumstances.

**Comment by mottosso** [comment]
I remember you had a stylesheet set-up?

**Comment by BigRoy** [comment]
Yes, but that was based on http://discourse.techart.online/t/release-qt-dark-orange-stylesheet/2287

Not sure if we can mimick that with the licensing we've got going. Especially if we change it here and there too.

**Comment by mottosso** [comment]
Don't worry about that, it doesn't apply here.

**Comment by BigRoy** [comment]
> Don't worry about that, it doesn't apply here.

How come?

**Comment by mottosso** [comment]
Because we aren't using it, we are building something that once drew inspiration from it. Still, in the worst of cases, what kind of repercussions are you expecting? That the author will sue? Let's include his name if you'd like. Or better yet, why not ask him for it?

**Comment by BigRoy** [comment]
Actually I just found the source repo I started from at the time: https://github.com/ColinDuquesnoy/QDarkStyleSheet

I really don't have much experience with styling Qt-applications and I'm not even sure whether the workflow in that repository is well suited for something like Avalon.

My "changed" version has the exact same code layout and only has some changed colors and padding here and there, they are more additional tweaks than a complete overhaul. That's why I also never used it aside from some personal apps here. So I think getting the licensing right (or at least us altering it without copyright issues) would be important too. Or to start from scratch, or at least redesign it largely.

**Comment by mottosso** [comment]
Ah, but that's even better, because that's actually got a license, MIT. :) So we're able to use it as-is, bundle it, ship it, distribute it, make a commercial product of it even. No limits!

**Comment by mottosso** [comment]
> I really don't have much experience with styling Qt-applications and I'm not even sure whether the workflow in that repository is well suited for something like Avalon.

No problem. I'm happy to help. CSS is a good route to standardising the look of applications. All we'd need is either take [this](https://github.com/ColinDuquesnoy/QDarkStyleSheet/blob/master/qdarkstyle/style.qss) file, and add another file for our customisations. We can use multiple files by simply adding them together..

```python
style1 = """QWidget {background: red}"""
style2 = """QWidget {border: 1px solid blue}"""
style = "\n".join([style1, style2])
```

And in that way we can keep our modifications separate from the original (and support theming!)

**Comment by BigRoy** [comment]
Actually, does that example code work? Doesn't the second QWidget declaration completely override the first? (Should just test this, hehe).

Anyhow, QDarkStyle also contains the encoded resources (the images) for the multiple Qt variants, like PyQt4, PySide, etc, which is convenient because we can easily use it in our toolsets. It could be a good candidate for vendorizing, also because it looks rather decent. We can always customize down the line to our own QAvalonStyle. Actually can we do that using QDarkStyle as basis (of course keeping the Licensing reference, etc.)?

**Comment by mottosso** [comment]
> Actually, does that example code work? Doesn't the second QWidget declaration completely override the first? (Should just test this, hehe).

That's the thing. They are cascading. It'd end up looking like this.

style.css

```bash
QWidget {background: red}
QWidget {border: 1px solid blue}
```

Which is fine.

What wouldn't work is this.

```python
app.setStyleSheet("""QWidget {background: red}""")
app.setStyleSheet("""QWidget {border: 1px solid blue}""")
```

Because *Qt* is the one overriding one with the other.

> Actually can we do that using QDarkStyle as basis (of course keeping the Licensing reference, etc.)?

You are much too worried about licensing! Yes, we can do absolutely everything with this library. Don't worry about it. If it helps, I officially take full responsibility for anything that happens using this library.

**Comment by BigRoy** [comment]
Implemented with #307 - the style itself is still open for discussion and can be improved along the way.
